### PR TITLE
Feature/patch gazelle and remove aws sdk replacements

### PR DESCRIPTION
**Pull request by stagnation**
We want the patches for gazelle to work with the 'aws-sdk' modules used
    by 'bb-storage'. Otherwise version resolution becomes much more
    tedious, with internal libraries used across the modules having
    incorrect references.

    One example problem is the s3 target, which fails here but succeeds in
    'bb-storage', with the gazelle patches.

        ERROR: /.../external/com_github_aws_aws_sdk_go_v2_service_s3
        /BUILD.bazel:3:11: no such target '@com_github_aws_aws_sdk_go_v2_internal_endpoints_v2//:awsrulesfn': tar
        get 'awsrulesfn' not declared in package '' defined by /CAS/bazel-cache/f1a15cfb78ca8fde7615be5d21224669/
        external/com_github_aws_aws_sdk_go_v2_internal_endpoints_v2/BUILD.bazel (Tip: use `query "@com_github_aws
        _aws_sdk_go_v2_internal_endpoints_v2//:*"` to see all the targets in that package) and referenced by '@co
        m_github_aws_aws_sdk_go_v2_service_s3//:s3'

    The difference in build files and dependencies:

        bb-storage $ bazel query 'deps(@com_github_aws_aws_sdk_go_v2_service_s3//:s3, 1)' \
            | grep awsrulesfn
        go_library rule @com_github_aws_aws_sdk_go_v2//internal/endpoints/awsrulesfn:awsrulesfn

        bb-deployments $ bazel query --output=build @com_github_aws_aws_sdk_go_v2_service_s3//:s3 \
            | linebreak-commas \
            | grep awsrulesfn
        "@com_github_aws_aws_sdk_go_v2_internal_endpoints_v2//:awsrulesfn",

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -26,8 +27,26 @@ http_archive(
     ],
 )
 
+# bb-storage is downloaded to separately to apply patches to gazelle.
+# Then the real `com_github_buildbarn_bb_storage` is downloaded through `go_dependencies.bzl`.
+git_repository(
+    name = "com_github_buildbarn_bb_storage_patches",
+    remote = "https://github.com/buildbarn/bb-storage",
+    commit = "ece87ab6dc2a9e1e592d2032f5a02c3694765cfc",
```
Comment (stagnation) [inline WORKSPACE:35]: This is now something we need to manage. :(

Comment (moroten) [reply WORKSPACE:34]: Should `http_archive` be used to download a source archive instead? If not, I'm happy for you to merge the PR.

Comment (stagnation) [reply WORKSPACE]: Good point! `http_archive` is better :)

Comment (stagnation) [reply WORKSPACE]: Done

**Pull request merged**